sdk/ml/azure-ai-ml/azure/ai/ml/_artifacts/_artifact_utilities.py

```diff
@@ -359,6 +370,125 @@ def _update_gen2_metadata(name, version, indicator_file, storage_client) -> None
     artifact_directory_client.set_metadata(_build_metadata_dict(name=name, version=version))
 
 
+def _generate_temporary_data_reference_id() -> str:
```

Comment:
For my understanding, why we need it?

Reply:
Do you mean why do we need this as a helper function or why does it need to be generated at all?

Reply:
Sorry for the confusion it is about why does it need to be generated at all,,,

Reply:
There has to be a unique name for the temporary data reference. Since this is an API call, the service isn't going to generate it itself, so until a contract is written and there's a method like the one you use for registry, we need to do it ourself here. If you look below, it's passed on as a required part of the API call.